File: WeatherForecast/WeatherForecast/Models/TitleValueModel.m
//
//  TitleValueModel.m
//  WeatherForecast
//
//  Created by Hakkı Yiğit Yener on 25.05.2020.
//  Copyright © 2020 N11.com. All rights reserved.
//

#import "TitleValueModel.h"

@implementation TitleValueModel

-(void)setValueWithNumber:(NSNumber *)value {
    self.value = [value stringValue];
}

+(instancetype)initWithTitle:(NSString*)title stringValue:(NSString*)value {
    TitleValueModel* model = [[TitleValueModel alloc] init];
    model.title = title;
    model.value = value;
    return model;
}
+(instancetype)initWithTitle:(NSString*)title numberValue:(NSNumber*)value {
    TitleValueModel* model = [[TitleValueModel alloc] init];
    model.title = title;
    model.value = [value stringValue];
    return model;
}

@end

File: WeatherForecast/WeatherForecast/Models/TitleValueModel.swift
//
//  TitleValueModelSwift.swift
//  WeatherForecast
//
//  Created by Hakkı Yiğit Yener on 27.05.2020.
//  Copyright © 2020 N11.com. All rights reserved.
//

import Foundation
class TitleValueModel: NSObject {
    var title: String?
    var value: String?

    func setValueWith(_ value: NSNumber?) {
        self.value = value?.stringValue ?? ""
    }
    init(title:String, stringValue value: String?) {
        self.title = title
        self.value = value
    }
    init(title:String, numberValue value: NSNumber?) {
        self.title = title
        self.value = value?.stringValue ?? ""
    }

}
